unstructured/utils.py

```diff
@@ -263,7 +263,7 @@ def validate_date_args(date: Optional[str] = None) -> bool:
     )
 
 
-def _first_and_remaining_iterator(it: Iterable[Any]) -> tuple[Any, Iterator[Any]]:
+def _first_and_remaining_iterator(it: Iterable[_T]) -> Tuple[_T, Iterator[_T]]:
     iterator = iter(it)
```

Comment:
@Coniferish did you notice these changes? This is a nice example of using a `TypeVar` to narrow `Any`.

The idea is that while it's true that `it` can be an iterable of any type, whatever type that is determines the output type(s). Like you can't send an iterable of `int` and expect `(str, Iterator[bool])` as a return value type, even though using Any for those would allow that. You should get `(int, Iterator[int])` in that case and this way the type-checker can verify that for you.

Kind of a cool example of generics in typing that I think is intuitive to understand.